pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,461 @@
+		require.Equal(t, expected, a)
+	case <-c:
+		t.Errorf("Expected event %q, got nothing", expected)
+		// continue iterating to print all expected events
+	}
+}
+
+func newTestPodConfigurator(testClients *mockClients, waiter *asyncWaiter) *podConfigurator {
+	interfaceStore := interfacestore.NewInterfaceStore()
+	eventBroadcaster := record.NewBroadcaster()
+	queue := workqueue.NewTypedDelayingQueueWithConfig[string](
+		workqueue.TypedDelayingQueueConfig[string]{
+			Name: "podConfigurator",
+		},
+	)
+	podCfg := &podConfigurator{
+		kubeClient:       testClients.kubeClient,
+		ofClient:         testClients.ofClient,
+		podLister:        testClients.podLister,
+		ifaceStore:       interfaceStore,
+		eventBroadcaster: eventBroadcaster,
+		recorder:         testClients.recorder,
+		unreadyPortQueue: queue,
+		containerAccess:  newContainerAccessArbitrator(),
+	}
+	if waiter != nil {
+		podCfg.podUpdateNotifier = waiter.notifier
+	}
+	return podCfg
+}
+
+func TestUpdateUnreadyPod(t *testing.T) {
+	defer mockRetryInterval()()
+
+	for _, tc := range []struct {
+		name               string
+		ofPortAssigned     bool
+		podIfaceIsCached   bool
+		installFlow        bool
+		flowInstalled      bool
+		installOpenFlowErr error
+		expErr             string
+		expEvent           string
+		expDeleted         bool
+	}{
+		{
+			name:             "updated Port is not in interface store",
+			podIfaceIsCached: false,
+			expDeleted:       true,
+			installFlow:      false,
+		}, {
+			name:             "OpenFlow port is not assigned",
+			podIfaceIsCached: true,
+			ofPortAssigned:   false,
+			installFlow:      false,
+			expDeleted:       false,
+			expErr:           "pod's OpenFlow port is not ready yet",
+		}, {
+			name:               "failed to install OpenFlow entries for updated Port",
+			podIfaceIsCached:   true,
+			ofPortAssigned:     true,
+			installFlow:        true,
+			installOpenFlowErr: fmt.Errorf("failure to install flow"),
+			expDeleted:         false,
+			expErr:             "failed to add Openflow entries for OVS port test: failure to install flow",
+			expEvent:           "Warning NetworkNotReady Pod network forwarding rules not installed",
+		}, {
+			name:               "succeeded",
+			podIfaceIsCached:   true,
+			ofPortAssigned:     true,
+			installFlow:        true,
+			installOpenFlowErr: nil,
+			expEvent:           "Normal NetworkReady Installed Pod network forwarding rules",
+			expDeleted:         true,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+
+			waiter := newAsyncWaiter(testPodNameA, podInfraContainerID)
+			defer waiter.close()
+
+			testClients := newMockClients(controller, nodeName, pod)
+			testClients.startInformers(stopCh)
+			fakeOFClient := testClients.ofClient
+
+			configurator := newTestPodConfigurator(testClients, waiter)
+
+			flowInstalled := false
+
+			ifConfig := podIfaceConfig
```

Comment:
I am not sure I understand the usage of `podIfaceConfig` in the tests. Why don't we just make it a pointer when we declare it and then use it directly instead of creating a `ifConfig` local variable?